File: src/main/java/com/abhinav/lms/ai/tutor/service/AiTutorService.java
package com.abhinav.lms.ai.tutor.service;

import com.abhinav.lms.ai.tutor.dto.ChatMessageResponse;
import com.abhinav.lms.ai.tutor.dto.ChatSessionResponse;
import com.abhinav.lms.ai.tutor.dto.CreateSessionRequest;
import com.abhinav.lms.ai.tutor.dto.SendMessageRequest;
import com.abhinav.lms.security.model.UserPrincipal;
import reactor.core.publisher.Flux;

import java.util.List;
import java.util.UUID;

public interface AiTutorService {

    ChatSessionResponse createSession(CreateSessionRequest request, UserPrincipal principal);

    List<ChatSessionResponse> getUserSessions(UserPrincipal principal);

    List<ChatSessionResponse> getUserSessionsByCourse(UUID courseId, UserPrincipal principal);

    ChatSessionResponse getSession(UUID sessionId, UserPrincipal principal);

    List<ChatMessageResponse> getSessionMessages(UUID sessionId, UserPrincipal principal);

    Flux<String> sendMessage(UUID sessionId, SendMessageRequest request, UserPrincipal principal);

    void deleteSession(UUID sessionId, UserPrincipal principal);
}

File: lms-frontend/services/aiTutorService.js
import api from './api';

export const aiTutorService = {
  async createSession(courseId = null, title = 'New Study Session') {
    const response = await api.post('/ai/tutor/sessions', {
      courseId,
      title
    });
    return response.data;
  },

  async getUserSessions(courseId = null) {
    const params = courseId ? { courseId } : {};
    const response = await api.get('/ai/tutor/sessions', { params });
    return response.data;
  },

  async getSessionMessages(sessionId) {
    const response = await api.get(`/ai/tutor/sessions/${sessionId}/messages`);
    return response.data;
  },

  async deleteSession(sessionId) {
    await api.delete(`/ai/tutor/sessions/${sessionId}`);
  }
};

export default aiTutorService;
